solr/modules/hdfs/src/java/org/apache/solr/hdfs/backup/repository/HdfsBackupRepository.java

```diff
@@ -204,6 +213,17 @@ public void copyIndexFileFrom(
     }
   }
 
+  @Override
```

Comment:
[-1] It's a downer that this implementation has to live in pretty much every BackupRepository implementation.  (Currently I think this is forced by the visibility of `shouldVerifyChecksum`, which is read by each individual BR implementation)

The BR interface has an `init(NamedList)` method - maybe we could read `verifyChecksums` in that base method, and avoid duplicating a bunch of this stuff on each implementation?

If it's not possible, fair enough, but it's worth trying before we commit this IMO

Reply:
I agree, this method should be shared, but BackupRepository is an interface, so I don't see how we share the sholdVerifyCheckum field. Unless we introduce an abstract class in-between, something like AbstractBackupRepository, it that what you mean?